linera-chain/src/data_types.rs

```diff
@@ -587,19 +587,71 @@ impl<'a> SignatureAggregator<'a> {
     }
 }
 
+fn is_ordered(values: &[(ValidatorName, Signature)]) -> bool {
+    values.windows(2).all(|pair| pair[0].0 <= pair[1].0)
+}
+
+fn has_duplicates(values: &[(ValidatorName, Signature)]) -> bool {
```

Comment:
At the call site we already know the vector is ordered, so we could do without the `HashSet`, and just:

```rust
    values.windows(2).all(|pair| pair[0].0 != pair[1].0)
```

But then why not just make `is_ordered` use `<` instead of `<=` and remove `has_duplicates`?

Reply:
(just reading this) +1 for `is_strictly_ordered` if this makes the code simpler